File: MVCApp/JAVA/MVCApp/src/Controller/Controller.java
package Controller;

import java.util.List;

import Controller.interfaces.iGetController;
import Controller.interfaces.iGetModel;
import Controller.interfaces.iGetView;
import Model.Model;
import Model.Domain.Student;
import View.View;

public class Controller implements iGetController {

    private iGetModel model;
    private iGetView view;
    private List<Student> students;

    public Controller(iGetModel model, iGetView view) {
        this.model = model;
        this.view = view;
    }

    private boolean test( List<Student> studs){
        if(studs.size()>0)
        {
            return true;
        }
        else
        {
            return false;
        }
    }

    public void update()
    {
        //MVP
        students = model.getAllStudents();

        if(test(students))
        {
            view.printAllStudent(students);
        }
        else{
            System.out.println("Модель не доступна!");
        }


        //MVC
        //Логика контроллера
        //view.printAllStudent(model.getAllStudents());
    }
    
    
}

File: MVCApp/python/Controller/controller.py
from typing import List
from Controller.Interfaces.i_get_controller import iGetController
from Controller.Interfaces.i_get_model import iGetModel
from Controller.Interfaces.i_get_view import iGetView
from Model.Domain.student import Student


class Controller(iGetController):
    def __init__(self, model: iGetModel, view: iGetView):
        self.model = model
        self.view = view
        self.students = []


    def test(self, studs: List[Student]) -> bool:
        if len(studs) > 0:
            return True
        else:
            return False

    def update(self):
        #MVP
        self.students = self.model.getAllStudents()

        if self.test(self.students):
            self.view.printAllStudent(self.students)
        else:
            print("Модель не доступна!")

        #MVC
        # Логика контроллера
        # self.view.printAllStudent(self.model.getAllStudents())